### ISSUE TITLE
Rethinking handling of `Wdmax`

### ISSUE THREAD

**Issue by yixinmao** [UW-Hydro/VIC#488]
Based on the conversation with @tbohn and @jhamman in PR #481 , `veg_var->Wdmax` is a derived model parameter (completely derived from `LAI`), and may not be ideal to be put in its current structure (currently it is handled as a state variable in `veg_var` structure). Need to rethink the handling of `Wdmax`.

**Comment by bartnijssen** [comment]
Agreed - should not be a state variable as it is a derived model parameter
